editor/src/document/movement_handler.rs

```diff
@@ -127,17 +128,27 @@ impl MessageHandler<MovementMessage, (&mut LayerData, &Document, &InputPreproces
 				responses.push_back(FrontendMessage::SetCanvasZoom { new_zoom: layerdata.scale }.into());
 				self.create_document_transform_from_layerdata(layerdata, &ipp.viewport_size, responses);
 			}
-			MultiplyCanvasZoom(multiplier) => {
-				let new = (layerdata.scale * multiplier).clamp(VIEWPORT_ZOOM_SCALE_MIN, VIEWPORT_ZOOM_SCALE_MAX);
-				layerdata.scale = new;
-				responses.push_back(FrontendMessage::SetCanvasZoom { new_zoom: layerdata.scale }.into());
-				self.create_document_transform_from_layerdata(layerdata, &ipp.viewport_size, responses);
+			IncreaseCanvasZoom => {
+				let next_level_higher = VIEWPORT_ZOOM_LEVELS.iter().find(|scale| **scale > layerdata.scale);
```

Comment:
not sure if it save lines in this case, but in general you can use scale = .find(…).unwrap_or(layerdata.scale); as a neat trick to remove the if let